www/include/TestResults.php

```diff
@@ -186,8 +186,17 @@ public function getMedianRunNumber($metric, $cached, $medianMode = "median") {
     if ($numValues == 1 || $medianMode == "fastest") {
       return $runNumbers[0];
     }
-    $medianIndex = (int)floor($numValues / 2.0);
-    return $runNumbers[$medianIndex];
+    $medianIndex = (int)floor(((float)$numValues + 1.0) / 2.0);
```

Comment:
Well it might also be simplified as `(int) floor(((float)$numValues - 1.0)/2.0)` to get a 0-based index.